internal/codegen/tpl_stencil_test.go

```diff
@@ -322,3 +327,41 @@ func TestTplStencil_ApplyTemplate(t *testing.T) {
 		})
 	}
 }
+
+func TestTplStencil_ReadDir(t *testing.T) {
+	log := slogext.NewTestLogger(t)
+	s := &TplStencil{
+		t: must(
+			NewTemplate(
+				must(modulestest.NewModuleFromTemplates(&configuration.TemplateRepositoryManifest{
+					Name: "test",
+				})),
+				"not-a-real-template.tpl",
+				0o755,
+				time.Now(),
+				[]byte{},
+				log,
+			),
+		),
+		s:   &Stencil{sharedState: newSharedState()},
+		log: log,
+	}
+
+	entries, err := s.ReadDir("../dsfdsfsd")
+	assert.Equal(t, true, errors.Is(err, billy.ErrCrossedBoundary))
+	assert.Equal(t, 0, len(entries))
+
+	entries, err = s.ReadDir("/root")
+	assert.Equal(t, true, errors.Is(err, os.ErrNotExist))
+	assert.Equal(t, 0, len(entries))
+
+	entries, err = s.ReadDir("testdata")
+	assert.NilError(t, err)
+	assert.Equal(t, true, len(entries) > 5)
+	assert.Equal(t, true, slices.ContainsFunc(entries, func(entry ReadDirEntry) bool {
+		return entry.Name == "blocks-test.txt" && !entry.IsDir
+	}))
+	assert.Equal(t, true, slices.ContainsFunc(entries, func(entry ReadDirEntry) bool {
+		return entry.Name == "args" && entry.IsDir
+	}))
```

Comment:
**thought:** This is kinda fragile if we add more files here ever

Reply:
i did > 5 just so it's got a stack and shouldn't need to touch it